### DISCUSSION TITLE
option to view currently time-tracked task and format time spent

### DISCUSSION THREAD

**Discussion by diagrammatics** [VladimirMarkelov/ttdl#93] [Ideas]
Currently, I have an alias to `ttdl -l tmr:1`, but it might be nice to have an option like "current" which reports the currently time-tracked task. 

Additionally, it might be nice to be able to format the spent field as a config option, i.e. "XXh XXm".

edit: Reading through the docs further, `--human-time=<format-string>` would be awesome and seems like it would conform to the existing option spec.

**Comment by VladimirMarkelov** [comment]
The option `--current` seems useful. It is possible that one may want to list all (stopped and active) or only stopped tasks. In this case `--current` is insufficient. Maybe, it needs a little more thinking over before adding a feature. How to name the option and how to use it? E.g, it can have optional value, without a value it works as your `tmr:1`, `-current=on` - the same, `--current=off` - only stoppped, `--current=all` - show both. It is a very raw idea.

As of time formatting, I have a question: where you want to see the formatted value? In a separate column or inside a subject. I guess, it is the former. Then another question: why the current formatting does not meet your criteria? When time is printed in its own column, it tends to print it as short as possible. Hence, it uses floating point instead of longer `HH:mm` format. Just an example:

```
# Content of todo.txt
first line spent:112 tmr:off
second line spent:8200 tmr:off

# List it with default options
> ttdl list 9,10
 # D P Created Finished Due Subject
------------------------------------
 9                          first line spent:112 tmr:off
10                          second line spent:8200 tmr:off

# List only non-empty fields
$ ttdl.exe list 9,10 --fields=spent,tmr
 # Spent  Tmr Subject
---------------------
 9 1.9m   off first line spent:112 tmr:off
10 2.3h   off second line spent:8200 tmr:off

# By hiding duplicates, we can get even better result
$ ttdl.exe list 9,10 --fields=spent,tmr --clean-subject=all

 # Spent  Tmr Subject
---------------------
 9 1.9m   off first line
10 2.3h   off second line
```

Yes, reading `1.9h` is harder than reading `52s`. But it is something that already exists. Another thing to think about is: let's assume we set format to `HH:MM`:

- What to display if time is only 6 second - the task just started?
- What if the total time exceeds 24 hours? `40h`?

Now let's choose format `DD HH MM`. The first questions from the previous list remains, and a new appear:
- Is `24h` always `1d` for time tracking? Some people may want to use business hours, e.g, `8h = = 1d`. Though, I think, in this case it may be OK to use `1d == 24h` always
- What to show in case of `5 h`? `0d 0h 0m` or `5h 0m` or `5h`? To me, `5h` is the best, but it a good way to over-complicate formatting part because it is so subjective.
